compiler/engine.go

```diff
@@ -19,7 +19,7 @@ type Engine interface {
 	// Compile defines a function that produces an executable
 	// representation of a pipeline from an object. This calls
 	// Parse internally to convert the object to a yaml configuration.
-	Compile(interface{}) (*pipeline.Build, error)
+	Compile(interface{}, *library.Build, *library.Repo) (*pipeline.Build, error)
```

Comment:
Could you elaborate a bit on why we're modifying the Compile function to accept a build and repo?

I ask, because currently the client for the compiler can be found here:

https://github.com/go-vela/compiler/blob/94875eae722b0342d410448257cf9c5c26e63333/compiler/native/native.go#L20-L30

You'll note that the client already has a `build` and `repo` field.

We then use a set of `With` functions to set these values in the client:

https://github.com/go-vela/compiler/blob/94875eae722b0342d410448257cf9c5c26e63333/compiler/native/native.go#L73-L80

https://github.com/go-vela/compiler/blob/94875eae722b0342d410448257cf9c5c26e63333/compiler/native/native.go#L109-L116

You can see where these function are called in [the `go-vela/server` codebase here](https://github.com/go-vela/server/blob/bd09d68c89acfae30dff29bd0ef58e0f7ac40d19/api/webhook.go#L364-L381):

```go
	// parse and compile the pipeline configuration file
	p, err := compiler.FromContext(c).
		WithBuild(b).
		WithComment(webhook.Comment).
		WithFiles(files).
		WithMetadata(m).
		WithRepo(r).
		WithUser(u).
		Compile(config)
	if err != nil {
		retErr := fmt.Errorf("%s: failed to compile pipeline configuration for %s: %v", baseErr, r.GetFullName(), err)
		util.HandleError(c, http.StatusInternalServerError, retErr)

		h.SetStatus(constants.StatusFailure)
		h.SetError(retErr.Error())

		return
	}
```

Reply:
> Could you elaborate a bit on why we're modifying the Compile function to accept a build and repo?

Simply did not realize that the client already had the required information available.. Will update my changes to utilize the client instead of passing the duplicate data.

Reply:
Yeah that's fair 😃 

I thought their was a potential for something larger at play that maybe I wasn't fully understanding.

At any rate, sounds like their is a plan forward so I'll be sure to keep eyes out for subsequent changes to this PR 🎉